### PR TITLE
Fix `INSTANCE_UPLOAD` action

### LINKED ISSUE #4524: Missing submission is not detected when user tries to send it via CollectTester
#### Software and hardware versions 
Collect master

#### Problem description
Problem with the missing submission is not recognized when user tries to send it via CollectTester

#### Steps to reproduce the problem
1. Fresh installations of the Collect and Collect Tester app are available on device
2. User opens Collect Tester
3. User goes to Submit Instance and put a random number
4. Sending form dialog is visible and the problem with missing submission is not recognized

#### Expected behavior
Dialog with error appears

| v1.30.1| master|
|---|---|
|Screenshot_20210419-154854|image|

#### Other information 
Related to #4439

### PR DESCRIPTION
**Pull request by seadowg**
Closes #4524
~~Blocked by #4534~~

As a failing test for this issue was better expressed using `ActivityScenario` rather than the deprecated rules, and because more and more tests are heading that way it seemed like a good time to switch away from `ActivityTestRule` and `IntentsTestRule`. This ended up being a surprisingly painless process after me assuming it was a spike I was going to have to bail out of. 

Given the changes I was making, @grzesiek2010 and I discussed also bundling in a change to have our tests start at the splash screen rather than the main menu. This is especially important now because we have a "first launch experience" that the tests were skipping (and was meaning we needed extra state setup).

#### What has been done to verify that this works as intended?

Ran tests.

#### Why is this the best possible solution? Were any other approaches considered?

Comments inline.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Should just change tests. No need for QA.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/external/AndroidShortcutsTest.java [hunk 1]
```diff
@@ -42,7 +45,10 @@ private FormEntryPage pickAndLaunchShortcutForForm(String formName) {
         Intent resultData = scenario.getResult().getResultData();
         Intent shortcutIntent = resultData.getParcelableExtra(Intent.EXTRA_SHORTCUT_INTENT);
 
-        rule.getActivity().startActivity(shortcutIntent);
-        return new FormEntryPage(formName, rule);
+         /*
+        This launch call doesn't work because of: https://github.com/android/android-test/issues/496
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/external/AndroidShortcutsTest.java:49]: I've created an issue that tracks the `android-test` issue for this.

File: collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.java [hunk 2]
```diff
@@ -78,12 +77,6 @@
 
 abstract class Page<T extends Page<T>> {
 
-    final ActivityTestRule rule;
-
-    Page(ActivityTestRule rule) {
```
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.java:83]: This is the change that is responsible for loads of other files changing. We'd previously needed `rule` to deal with getting an Activity for string translation but weren't using it anymore. This means that our pages can work without an `ActivityTestRule` and means that we can now use `AcitivityScenario` instead.

File: collect_app/src/test/java/org/odk/collect/android/activities/MainMenuActivityTest.kt [hunk 3]
```diff
@@ -33,20 +35,20 @@ import org.odk.collect.projects.Project
...
-    private lateinit var currentProject: Project
-    private lateinit var livedata: LiveData<Int>
+    private val project = Project("Project", "P", "#f5f5f5")
 
-    @Before
-    fun setup() {
-        mainMenuViewModel = mock(MainMenuViewModel::class.java)
-        currentProject = mock(Project::class.java)
-        livedata = mock(LiveData::class.java) as LiveData<Int>
+    private val mainMenuViewModel = mock<MainMenuViewModel> {
```
Comment (seadowg) [inline collect_app/src/test/java/org/odk/collect/android/activities/MainMenuActivityTest.kt:40]: @grzesiek2010 this is mockito-kotlin

**Review by lognaturel — DISMISSED** [review]
What has happened I ended up on totally the wrong PR.

**Review by grzesiek2010 — APPROVED** [review]

**Comment by mmarciniak90** [comment]
**Tested with success**

Verified on Android 7.0, 8.1, 10.0

**Verified cases:**
- submit instance which does not exist
- submit an existing instance
- download form which does not exist
- download an existing form
- use default server from Collect
- use server specified in ODK Tester
- dialog with missing credentials

**Comment by kkrawczyk123** [comment]
Verified also on Androids 5.1, 9.0 and 11.0.

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**